### PR TITLE
feat(#238): add per-pet litter box bathroom activity sensors

### LINKED ISSUE #238: [Feature Request]: Luma Bathroom history
### Device model name

Luma

### Device model number

PLLB001

### Device type

Other

### Feature name

Record Litterbox Usage Data for Pets

### Describe the feature

The Petlibro Luma litter box provides history on linked pets. Recording this usage and waste type can be very useful in understanding pet behavior and assist in identifying medical conditions. 

### What problem does this feature solve?

Review pet bathroom usage patterns

### Available to help with the feature development

- [x] I agree to help with testing or providing relevant information for this feature request

### PR DESCRIPTION
**Pull request by menardza**
Addresses #238.

Adds four new per-pet sensors for Luma Smart Litter Box users:
- Today's Bathroom Visits
- Today's Bathroom Duration
- Today's Poop Count
- Today's Pee Count

Data comes from /data/pet/potty/today, which returns all bound pets' data in a single response (similar to wearListV2 for fountains) - the petId in the request just needs to be non-empty, it does not filter server-side. Cached per device serial to avoid redundant calls when multiple pets share one litter box.

Tested against a live Luma Smart Litter Box (PLLB001).

Known limitation: since pets refresh concurrently via asyncio.gather, the per-device cache doesn't always dedupe calls within the same refresh cycle (observed up to 5 calls/minute for a 5-pet household instead of 1). Data returned is correct either way - this is a minor efficiency follow-up, not a correctness issue. Happy to address in a follow-up commit if wanted.

## Proposed change:

Adds per-pet litter box bathroom activity tracking (visits, duration, poop/pee counts) for Luma Smart Litter Box owners, following the same per-pet sensor pattern already used for RFID fountain drinking data. English translations included for the new sensors.

Closes #238

## Type of change:

- [ ] New device.
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature or enhancement (non-breaking change which adds functionality).
- [ ] Documentation only.
- [ ] Other (please explain).

## Checklist:

- [ ] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature / enhancement](https://github.com/jjjonesjr33/petlibro/wiki/Feature-&-Enhancement-Guidelines) guidlines before submitting my request.
- [x] If applicable, I have tested my code for new features & regressions on the latest version of Home Assistant.

## Additional notes:

This covers today's summary only (visits, duration, poop/pee counts), which is what was requested in #238. It does not add support for /data/petCare/bathroom/history (the longer-range trend endpoint also mentioned in the original issue) - I tested it, but every request returned an empty shape (averageTimes: 0, averageDuration: 0, vetRecommendation: null, community: null, ydata: [], xdate: []) regardless of pet or device, suggesting it needs additional parameters (likely a date range) that I wasn't able to determine.

### CODE REVIEW

File: custom_components/petlibro/api.py [hunk 1]
```diff
@@ -545,6 +545,20 @@ async def device_data_real_info(self, serial: str) -> Dict[str, Any]:
     async def device_drink_water(self, serial: str) -> Dict[str, Any]:
         return await self.session.post_serial("/data/deviceDrinkWater/todayDrinkData", serial)
 
+    async def device_potty_today(self, serial: str, pet_id: int) -> Dict[str, Any]:
+        # Cached by device serial only (not pet_id) — this endpoint returns ALL
+        # pets' data in one response regardless of which pet_id is sent; pet_id
+        # just needs to be non-empty. Caching avoids calling once per pet.
+        now = utcnow()
+        cache_key = f"{serial}_pottyToday"
+        last_call_time = self._last_api_call_times.get(cache_key)
```
Comment (jjjonesjr33) [inline custom_components/petlibro/api.py:557]: @menardza I want to get the concurrent request issue fixed before we merge this. Since the pets are refreshing at the same time with `asyncio.gather`, they can all get past the cache check before the first request finishes and updates the cache. That's why we're seeing up to 5 calls with 5 pets instead of the 1 call we want.

Let's track the request that's already running for each `cache_key`. If there's already a request running for that device, the other pet refreshes should wait for that same request instead of making another API call.

Example:
```python
# Add to the API class
self._inflight_requests: dict[str, asyncio.Task] = {}


async def device_potty_today(
    self, serial: str, pet_id: int
) -> Dict[str, Any]:
    cache_key = f"{serial}_pottyToday"
    now = utcnow()

    last_call_time = self._last_api_call_times.get(cache_key)
    if last_call_time and (now - last_call_time) < timedelta(seconds=10):
        return self._cached_responses.get(cache_key, {})

    if cache_key in self._inflight_requests:
        return await self._inflight_requests[cache_key]

    async def _fetch() -> Dict[str, Any]:
        result = await self.session.post_serial(
            "/data/pet/potty/today",
            serial,
            json={"petId": pet_id},
        )
        self._last_api_call_times[cache_key] = utcnow()
        self._cached_responses[cache_key] = result
        return result

    task = asyncio.create_task(_fetch())
    self._inflight_requests[cache_key] = task

    try:
        return await task
    finally:
        self._inflight_requests.pop(cache_key, None)
```
That should keep the pets refreshing concurrently while getting `/data/pet/potty/today` down to one request per device instead of potentially making a request for every pet.

**Review by jjjonesjr33 — CHANGES_REQUESTED** [review]
